### PR TITLE
Move cellStatus to cellMap

### LINKED ISSUE #713: Move cellStatuses attributes to cells under cellMap
The attributes stored in the top-level cellStatuses object should be stored in each cell at the cellMap level.

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #713.

### CODE REVIEW

**Comment by rgbkrk** [comment]
We'll still need to remove these from the saved notebook too, otherwise jupyter won't be able to open it (probably worth checking on this).

**Comment by captainsafia** [comment]
> We'll still need to remove these from the saved notebook too, otherwise jupyter won't be able to open it (probably worth checking on this).

Oh drat! Totally forgot about that. Let me fix that.

**Comment by rgbkrk** [comment]
> Oh drat! Totally forgot about that. Let me fix that.

All good. Probably worth noting that I _think_ you can do this in nteract, making sure to use the same function in both saveNotebook and publishing.

**Comment by captainsafia** [comment]
OK. Updated. Although we probably wanna move this code to a utility function within commutable.

**Comment by rgbkrk** [comment]
I actually forsee us having other changes to the in-memory model here that we may want to keep experimenting with them here until we're sure about changing the interface in commutable.

File: src/notebook/epics/github-publish.js [hunk 1]
```diff
@@ -134,7 +134,12 @@ export function createGistCallback(firstTimePublish, observer, filename, notific
  */
 export function publishNotebookObservable(github, notebook, filepath, notificationSystem) {
   return Rx.Observable.create((observer) => {
-    const notebookString = JSON.stringify(commutable.toJS(notebook), undefined, 1);
+    const notebookString = JSON.stringify(
+      commutable.toJS(notebook.update('cellMap', cells =>
+        cells.map(value =>
+          value.delete('inputHidden').delete('outputHidden').delete('status')))),
+      undefined,
```
Comment (rgbkrk) [inline src/notebook/epics/github-publish.js:141]: Yeah, I'd actually move this code to our own `toJS` helper, maybe with a file named:

`notebook-model.js`
`notebook-memory-model.js`

Ok, I'm out of ideas for naming here.

**Pull request merged**